src/Microsoft.AspNet.Mvc.Core/Formatters/HttpNotAcceptableOutputFormatter.cs

```diff
@@ -1,40 +0,0 @@
-// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System;
-using System.Collections.Generic;
-using System.Threading.Tasks;
-using Microsoft.AspNet.WebUtilities;
-using Microsoft.Net.Http.Headers;
-
-namespace Microsoft.AspNet.Mvc
-{
-    /// <summary>
-    /// A formatter which does not have a supported content type and selects itself if no content type is passed to it.
-    /// Sets the status code to 406 if selected.
-    /// </summary>
-    public class HttpNotAcceptableOutputFormatter : IOutputFormatter
```

Comment:
Ok, some scenarios which drove me to do this are:
- Disable match on object type application wide (this can be done at MvcOptions level)
- Let the default behavior be as it is but for certain controllers/actions, disable match on object type (this can be done through the property on ObjectResult)

I do not see any more interesting scenarios...so I do not think this formatter needs to be present...thoughts? @rynowak